Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -15,15 +15,42 @@ struct GHOAuthHandler: LambdaHandler {
     let client: HTTPClient
     let logger: Logger
     let secretsRetriever: SecretsRetriever
+    let discordClient: any DiscordClient
 
-    init(context: LambdaInitializationContext) {
+    init(context: LambdaInitializationContext) async throws {
         self.client = HTTPClient(eventLoopGroupProvider: .shared(context.eventLoop))
         self.logger = context.logger
+
         let awsClient = AWSClient(httpClientProvider: .shared(client))
         self.secretsRetriever = SecretsRetriever(awsClient: awsClient, logger: logger)
+
+        let botToken = try await secretsRetriever.getSecret(arnEnvVarKey: "BOT_TOKEN_ARN")
+        self.discordClient = await DefaultDiscordClient(httpClient: client, token: botToken)
+    }
+
+    func handle(_ event: APIGatewayV2Request, context: LambdaContext) async throws -> APIGatewayV2Response {
```

Comment:
😅 This will still now work well for us. You can keep the function as `throws`, but should try to actually not throw unless necessary:

So the difference between the `GHHooks` lambda and this lambda is that the other side of `GHHooks` lambda is Github who sends the events, but the other side of this lambda is an actual human.

With throwing anywhere, the lambda will show a 500 error to user with no description of what has happened. The lambda won't mention the error description etc...

That's why the `handle` function should not throw, unless absolutely necessary.
Throwing e.g. in the initializer when a secret or env var is not available, is fine. Because without a secret we can't continue our work, and not-finding a secret is an actual 500 error anyway since something in our code is not working.

But other than that you should try to not throw at all, unlike the GHHooks lambda (or if something needs to throw, use a do/catch instead)